edge/pkg/edged/edged_pods.go

```diff
@@ -530,3 +532,334 @@ func (e *edged) removeOrphanedPodStatuses(pods []*v1.Pod) {
 
 	e.statusManager.RemoveOrphanedStatuses(podUIDs)
 }
+
+
+// GetPodCgroupParent gets pod cgroup parent from container manager.
+func (e *edged) GetPodCgroupParent(pod *v1.Pod) string {
+	/*pcm := e.containerManager.NewPodContainerManager()
+	_, cgroupParent := pcm.GetPodContainerName(pod)
+	return cgroupParent*/
+	return "systemd"
+}
+
+
+// GenerateRunContainerOptions generates the RunContainerOptions, which can be used by
+// the container runtime to set parameters for launching a container.
+func (e *edged) GenerateRunContainerOptions(pod *v1.Pod, container *v1.Container, podIP string) (*kubecontainer.RunContainerOptions, func(), error) {
+	/*opts, err := e.GenerateContainerOptions(pod)
+	if err != nil {
+		return nil, nil, err
+	}*/
+	opts := kubecontainer.RunContainerOptions{}
+
+
+	hostname, hostDomainName, err := e.GeneratePodHostNameAndDomain(pod)
+	if err != nil {
+		return nil, nil, err
+	}
+	opts.Hostname = hostname
+	podName := util.GetUniquePodName(pod)
+	volumes := e.volumeManager.GetMountedVolumesForPod(podName)
+	opts.PortMappings = kubecontainer.MakePortMappings(container)
+
+	// TODO: remove feature gate check after no longer needed
+	if utilfeature.DefaultFeatureGate.Enabled(features.BlockVolume) {
+		blkutil := volumepathhandler.NewBlockVolumePathHandler()
+		blkVolumes, err := e.makeBlockVolumes(pod, container, volumes, blkutil)
+		if err != nil {
+			return nil, nil, err
+		}
+		opts.Devices = append(opts.Devices, blkVolumes...)
+	}
+
+	/*envs, err := e.makeEnvironmentVariables(pod, container, podIP)
+	if err != nil {
+		return nil, nil, err
+	}
+	opts.Envs = append(opts.Envs, envs...)*/
+
+	mounts, err := makeMounts(pod, e.getPodDir(pod.UID), container, hostname, hostDomainName, podIP, volumes)
+	if err != nil {
+		return nil, nil, err
+	}
+	opts.Mounts = append(opts.Mounts, mounts...)
+
+	// Disabling adding TerminationMessagePath on Windows as these files would be mounted as docker volume and
+	// Docker for Windows has a bug where only directories can be mounted
+	if len(container.TerminationMessagePath) != 0 && runtime.GOOS != "windows" {
+		//p := e.getPodContainerDir(pod.UID, container.Name)
+		p := e.getPodContainerDir(pod.UID, container.Name)
+		if err := os.MkdirAll(p, 0750); err != nil {
+			glog.Errorf("Error on creating %q: %v", p, err)
+		} else {
+			opts.PodContainerDir = p
+		}
+	}
+
+	return &opts, nil, nil
+}
+// GetPodDNS returns DNS settings for the pod.
+// This function is defined in kubecontainer.RuntimeHelper interface so we
+// have to implement it.
+func (e *edged) GetPodDNS(pod *v1.Pod) (*runtimeapi.DNSConfig, error) {
+	dnsConfig := &runtimeapi.DNSConfig{Servers:[]string{"",},}
+	return dnsConfig, nil
+}
+
+
+// Make the environment variables for a pod in the given namespace.
```

Comment:
Code required...will be uncommented later....to create issue for same

Reply:
Please add additional comment to explain why this piece of code is needed in comment by now, and the criteria of uncommenting it. Otherwise please remove it.